pkg/alertmanager/alertmanager.go

```diff
@@ -137,7 +137,13 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.state = cfg.Peer
 	} else if cfg.ShardingEnabled {
 		level.Debug(am.logger).Log("msg", "starting tenant alertmanager with ring-based replication")
-		am.state = newReplicatedStates(cfg.UserID, cfg.ReplicationFactor, cfg.ReplicateStateFunc, cfg.GetPositionFunc, am.stop, am.logger, am.registry)
+		state := newReplicatedStates(cfg.UserID, cfg.ReplicationFactor, cfg.ReplicateStateFunc, cfg.GetPositionFunc, am.logger, am.registry)
+
+		if err := state.Service.StartAsync(context.Background()); err != nil {
+			return nil, fmt.Errorf("failed to start state replication service: %v", err)
```

Comment:
It's preferable to wrap errors.

```suggestion
			return nil, errors.Wrap(err, "failed to start state replication service")
```

Reply:
Sure - only reason I didn't was because the rest of the file does not use `Wrap`, so preferred consistency.

Reply:
I think most code here is pretty old and we're not trying to convert to `errors.Wrap()`.